### PR TITLE
WEBEU2 - JavaScript II - Josemaria Nriagu 🇳🇬

### PR DESCRIPTION
**Pull request by josenriagu**
Started project

### CODE REVIEW

File: assignments/array-methods.js [hunk 1]
```diff
@@ -2,82 +2,119 @@
...
+    return {'id': name.id, 'first_name': name.first_name.toUpperCase(), 'last_name': name.last_name, 'email': name.email, 'shirt_size': name.shirt_size, 'company_name': name.company_name, 'donation': name.donation} */
+    name.first_name = name.first_name.toUpperCase();
+    return name;
+});
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter((size) => {
```
Comment (maximesalomon) [inline assignments/array-methods.js]: 👍

File: assignments/array-methods.js [hunk 2]
```diff
@@ -2,82 +2,119 @@
...
+    return size.shirt_size === 'L';
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((holder, element) => {
```
Comment (maximesalomon) [inline assignments/array-methods.js]: holder is called an accumulator

File: assignments/array-methods.js [hunk 3]
```diff
@@ -2,82 +2,119 @@
...
+    return holder + element.donation
+}, 0)
+
+console.log(ticketPriceTotal2);
+
+// Problem 3 - return an array of donors whose first name start with a letter 'J'
 
-// Problem 2
+let jNames = runners.filter((element) => {
+    return element.first_name.split('')[0] === 'J'
```
Comment (maximesalomon) [inline assignments/array-methods.js]: Great job using the split array method

File: assignments/callbacks.js [hunk 4]
```diff
@@ -24,32 +24,73 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
+
 /* STRETCH PROBLEM */
+const test = [4,1,1,3,3,2,3,4,2,5,5,6,6,6];
 
 function removeDuplicates(array, cb) {
   // removeDuplicates removes all duplicate values from the given array.
   // Pass the duplicate free array to the callback function.
   // Do not mutate the original array.
+  let noDupes = array.filter((item, index) => {
+    return array.indexOf(item) === index;
```
Comment (maximesalomon) [inline assignments/callbacks.js]: Nice one!

**Comment by josenriagu** [comment]
Thanks a lot Max. I'll take note of the corrections. I hope my commits
today are better

Regards,

Josemaria Nriagu
🇳🇬

**Pull request merged**